decorator.rb

```diff
@@ -0,0 +1,11 @@
+require_relative 'nameable'
+class BaseDecorator < Nameable
+  def initialize(namable)
+    super()
+    @namable = namable
+  end
+
+  def correct_name
+    @namable.correct_name
```

Comment:
- [ ] Please add the `getter` attribute for the `@namable` instance variable because by doing this you will then be able to expose the `value` or make the `@namable` readable outside the class using the object.